src/lib/yang/schema.lua

```diff
@@ -700,7 +707,10 @@ function resolve(schema, features)
                   node.body[k] = v
                end
                -- TODO: Handle refine and augment statements.
-            else
+            end
+         end
+         for k,v in pairs(node.body or {}) do
```

Comment:
Interesting.

I note first that after the body has been visited once (as it is above), there will be no more "uses" nodes -- they will be replaced.

I wonder if your interpretation of "uses" is correct.  Other things in YANG are scoped lexically -- for example the set of available typedefs.  But you are suggesting that "config" is instead scoped dynamically.  That is why you are having to go back and revisit the nodes -- you put "config" in the lexical environment `env` but really (if you are right) we need to have a dynamic environment.

If you are right -- and I think I would like to see a schema that shows this usage -- then I think a better plan is do propagate "config" in a pass that runs after resolve.  Otherwise things are quadratic, as revisiting each tree level will recursively revisit all levels underneath.